### PR TITLE
Shooting functionality

### PR DESCRIPTION
**Pull request by filiptrplan**
* added shooting functionality to backend and frontend
* the only caveat is that we still don't have a backend physics update loop, so the bullets don't get deleted from the server array --> I disabled this functionality until we have a physics loop on the server to check if the bullet should be deleted
* this was a pain to refactor but it now works perfectly :)
* **ALSO** changed the base Object class to generate a random crypto string id so we don't have to specify id each time we create an object

### CODE REVIEW

File: world-client/src/features/shooting.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import Feature from "./feature";
+import * as Phaser from 'phaser';
+import Player from "../state/player";
+export default class Shooting extends Feature {
+
+    constructor(){
```
Comment (jakakonda) [inline world-client/src/features/shooting.js:6]: minor: Formatting.

**Review by davidmohar — APPROVED** [review]
Approved for now but I recommend we move the shooting functionality into it's own handler. It can still be player scoped.

File: world-client/src/features/shooting.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        );
+        wop.state.state.bullets.push(bullet);
+        break;
+    }
+  }
+
+}
+
+export class Bullet {
+    constructor(id, damage, BulletSpeed, lifetime, pos={x: wop.me.character.x, y: wop.me.character.y}, vel={x:0,y:0}, angle=null){
```
Comment (jakakonda) [inline world-client/src/features/shooting.js:90]: minor, naming convention. Variables start with lower case.

File: world-server/src/world/player/PlayerHandlers.ts [hunk 3]
```diff
@@ -82,6 +85,13 @@ export class PlayerHandler implements Messages.MsgHandler {
                 this.players.splice(index, 1);
                 Router.emit(new BroadcastMessage(message));
                 break;
+            
+            case 'player.shoot':
```
Comment (jakakonda) [inline world-server/src/world/player/PlayerHandlers.ts:89]: I'd suggest moving bullets to a separate handler for readability and manageability.

**Review by jakakonda — APPROVED** [review]
Approved. Comments should be addressed in the near future.

**Pull request merged**